src/main/java/com/exadel/discount/controller/CategoryController.java

```diff
@@ -0,0 +1,55 @@
+package com.exadel.discount.controller;
+
+import com.exadel.discount.dto.CategoryDTO;
+import com.exadel.discount.dto.validation.Create;
+import com.exadel.discount.service.interfaces.CategoryService;
+import io.swagger.annotations.ApiOperation;
+import lombok.RequiredArgsConstructor;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@RequiredArgsConstructor
+@Validated
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+
+    private final CategoryService categoryService;
+
+    @GetMapping
+    @ApiOperation("Get all categories or category by id")
+    public List<CategoryDTO> getAll(@RequestParam(name = "id", required = false) UUID id) {
+        if (id == null) {
```

Comment:
I don't think that combining getAll and getById is a good idea. The name of your method is getAll, it's not good to find by id in it. And also for other entities the same. One more reason to devide these methods is, for example, for discounts: in getAll we'll pass some sorting, paging parameters

Reply:
+